tools/toolchain/clang_lto_pgo.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash -uex
+
+# Which Scylla branch to train on
+SCYLLA_BRANCH=master
+
+# Which LLVM release to build
+LLVM_TAG=llvmorg-15.0.7
+
+if [ -d "/optimized_clang" ];
+then
+	DIR="/optimized_clang"
+else
+	DIR="${PWD}/optimized_clang"
+	mkdir -p "${DIR}"
+fi
+cd "${DIR}"
+
+# Download the scylla codebase for training.
+# Scylla's dependencies are already installed.
+git clone https://github.com/scylladb/scylla --branch ${SCYLLA_BRANCH} --depth=1
+git -C scylla submodule update --init --depth=1
+
+# Clone, patch and bootstrap the newest Clang and BOLT.
+#git clone https://github.com/llvm/llvm-project --branch main --depth=1 stage-0
+git clone https://github.com/llvm/llvm-project --branch main --depth=1 stage-0
+cd stage-0
+
+cmake -B build -S llvm -DLLVM_ENABLE_PROJECTS='clang;lld;bolt' -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD=X86 -G Ninja -DLLVM_ENABLE_RUNTIMES='compiler-rt' -DCOMPILER_RT_BUILD_SANITIZERS=Off -DCOMPILER_RT_DEFAULT_TARGET_ONLY=On -DLLVM_INCLUDE_BENCHMARKS=No -DLLVM_INCLUDE_EXAMPLES=No -DLLVM_INCLUDE_TESTS=No
+cmake --build build --
+
+USE_NEW_COMPILER=(-DCMAKE_C_COMPILER="$DIR/stage-0/build/bin/clang" -DCMAKE_CXX_COMPILER="$DIR/stage-0/build/bin/clang++" -DLLVM_USE_LINKER="$DIR/stage-0/build/bin/ld.lld")
+COMMON_OPTS=(-DLLVM_ENABLE_PROJECTS='clang' -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD=X86 -G Ninja -DLLVM_ENABLE_LTO=Thin -DLLVM_BUILD_RUNTIME=No -DCLANG_DEFAULT_PIE_ON_LINUX=OFF -DLLVM_INCLUDE_BENCHMARKS=No -DLLVM_INCLUDE_EXAMPLES=No -DLLVM_INCLUDE_TESTS=No)
```

Comment:
Let's configure the compiler to install into /opt, and link the compiler statically, so people can use it outside dbuild.